src/renderer/components/inputs/InputContainer.tsx

```diff
@@ -1,64 +1,208 @@
+            position={Position.Left}
+            type="target"
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...props}
+        >
+            {children}
+        </Handle>
+    )
+);
+
+const Div = chakra('div', {
+    baseStyle: {},
+});
+
 const InputContainer = memo(
-    ({ children, hasHandle, id, inputId, label }: React.PropsWithChildren<InputContainerProps>) => {
-        const { isValidConnection } = useContext(GlobalVolatileContext);
+    ({
+        children,
+        hasHandle,
+        id,
+        inputId,
+        label,
+        type,
+    }: React.PropsWithChildren<InputContainerProps>) => {
+        const { isValidConnection, edgeChanges, useConnectingFromType, useConnectingFrom } =
+            useContext(GlobalVolatileContext);
+        const { getEdges, getNode } = useReactFlow();
+        const edges = useMemo(() => getEdges(), [edgeChanges]);
+        const connectedEdge = edges.find(
+            (e) => e.target === id && parseHandle(e.targetHandle!).inOutId === inputId
+        );
+        const isConnected = !!connectedEdge;
+        const [connectingFromType] = useConnectingFromType;
+        const [connectingFrom] = useConnectingFrom;
+
+        const showHandle = useMemo(() => {
+            if (
+                !connectingFrom ||
+                !connectingFromType ||
+                // We want to display the connectingFrom handle
+                (connectingFrom.handleId === `${id}-${inputId}` &&
+                    connectingFrom.handleType === 'target')
+            ) {
+                return true;
+            }
+            // If the connecting from node is the same as the node we're connecting to
+            if (connectingFrom.nodeId === id) {
+                return false;
+            }
+            // Any other inputs should be invalid
+            if (connectingFrom.handleType === 'target') {
+                return false;
+            }
+            // Show same types
+            const connectionIsValid = isValidConnection({
+                source: connectingFrom.nodeId,
+                sourceHandle: connectingFrom.handleId,
+                target: id,
+                targetHandle: `${id}-${inputId}`,
+            });
+            if (connectionIsValid && intersect(connectingFromType, type).type !== 'never') {
+                return true;
+            }
+            return false;
+        }, [connectingFrom, connectingFromType, type, id, inputId]);
+
+        const { typeDefinitions, functionDefinitions } = useContext(GlobalContext);
+        const { useIsDarkMode } = useContext(SettingsContext);
+        const [isDarkMode] = useIsDarkMode;
 
         let contents = children;
-        const handleColor = useColorModeValue('#EDF2F7', '#171923');
-        const borderColor = useColorModeValue('#171923', '#F7FAFC');
+        const handleColors = getTypeAccentColors(type, typeDefinitions, isDarkMode); // useColorModeValue('#EDF2F7', '#171923');
```

Comment:
Still more commented out code. I think I saw this in a few other places as well. Maybe search with the regex pattern `// [a-z]\w*[(=:]` in the whole code base to find them all?